File: 4365-count-valid-prefixes/count-valid-prefixes.py
class Solution:
    def countValidPrefixes(self, s: str) -> int:
        count0 = count1 = ans = 0

        for ch in s:
            if ch == "0":
                count0 += 1
            else:
                count1 += 1

            if abs(count1 - count0) <= 1:
                ans += 1

        return ans

File: 4365-count-valid-prefixes/count-valid-prefixes.go
func countValidPrefixes(s string) int {
	count0, count1, ans := 0, 0, 0

	for _, ch := range s {
		if ch == '0' {
			count0++
		} else {
			count1++
		}

		diff := count1 - count0
		if diff < 0 {
			diff = -diff
		}

		if diff <= 1 {
			ans++
		}
	}

	return ans
}